### ISSUE TITLE
Block Template Overriding 

### ISSUE THREAD

**Issue by munirkamal** [AdvancedCustomFields/acf#231]
Hi @elliotcondon,

I am looking for a way / best practice to allow users to override default Block templates come with my ACFBlocks plugin. So, for example, I have a testimonial acf block which has a template comes with my plugin, but I want to let a theme developer/user override the template easily by adding the template file/folder to the theme directory and the testimonial block (or any block) prioritize the theme folder to pick the template instead of one available in the plugin itself. 

Any hints/direction would be really helpful.

Note: Also planning to add multiple block styles that may pick different templates, that to be explored. Any hints for that would be great too.

TIA

**Comment by munirkamal** [comment]
I think something like overriding the template path filter as Block Lab provides would be a solution? Is there already a filter for that for us in ACF? or a better way may be to handle the path directly by recognizing the specific folder similarly to how EDD/WOO allows to override their templates. 
https://getblocklab.com/docs/get-started/add-a-block-lab-block-to-your-website-content/

I am just thinking about it. 😀
Please suggest the best way to do it.

Thanks

**Comment by elliotcondon** [comment]
Hi @munirkamal 

I would also consider the approach taken by WooCommerce and their theme templates.
WC has many templates for the theme HTML and these can be overridden by creating a folder "woocommerce" in the current theme and copying in the desired template files.

You could easily add this functionality into your `render_callback` function, or even when registering the block during `acf/init`.

**Comment by munirkamal** [comment]
@elliotcondon 

Yeah, I am thinking to adopt the same approach as well. I am kind of struggling to put the pieces of this puzzle. Is there any example of how to make it happen? Or any hint to help me get on the right direction would be really appreciated.

TIA

**Comment by elliotcondon** [comment]
@munirkamal 

I can see two solutions here:

1. During "acf/init" action, check for template file overrides when registering the block, and use the final template file in the `render_template` setting.

2. During the `render_callback` function, look for template overrides using the same logic.

Either way, you will need to write some code to search through the active theme and look for theme overrides. The logic is up to you. Checkout Google and Stackoverflow for code examples on searching through file systems.
